contrib/metrics/supply_check/supply_deltas.py

```diff
@@ -0,0 +1,58 @@
+import pprint
+import bisect
+
+class SupplyDeltas:
+    def __init__(self, zcashd, fr_addrs, miner_deltas, flush_interval = 500):
+        self.zcashd = zcashd
+        self.fr_addrs = fr_addrs
+        self.miner_deltas = miner_deltas
+
+        self.delta_total = 0
+        self.delta_cache = []
+        self.flush_interval = flush_interval
+        
+        deltas_flat = [pair for deltas in miner_deltas.values() for pair in deltas]
+        for (deltaHeight, delta) in sorted(deltas_flat):
+            self.AddSupplyDelta(deltaHeight, delta)
+
+    def AddSupplyDelta(self, deltaHeight, delta):
+        assert len(self.delta_cache) == 0 or deltaHeight > self.delta_cache[-1][0]
+
+        self.delta_total += delta
+        self.delta_cache.append((deltaHeight, self.delta_total))
```

Comment:
In order to use `bisect.insort` it would be necessary to keep a separate running cache of delta totals, and recompute that cache each time a value is inserted out of order, so I think that maintaining the insertion-order requirement (and the related assertion) is the better solution.